omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/PatientIdentifierController1_9Test.java

```diff
@@ -114,6 +115,33 @@ public void shouldEditAPatientIdentifier() throws Exception {
 	}
 	
 	@Test
+	public void shouldUnsetOtherPreferredIdentifiers() throws Exception {
+		PatientIdentifier nonPreferred = service.getPatientIdentifierByUuid(getUuid());
+		long originalCount = getAllCount();
+		assertTrue(nonPreferred.isPreferred());
+		SimpleObject patientIdentifier = new SimpleObject();
+		patientIdentifier.add("identifier", "abc123ez");
+		patientIdentifier.add("identifierType", "2f470aa8-1d73-43b7-81b5-01f0c0dfa53c");
+		patientIdentifier.add("location", RestTestConstants1_8.LOCATION_UUID);
+		patientIdentifier.add("preferred", true);
+		
+		String json = new ObjectMapper().writeValueAsString(patientIdentifier);
+		
+		MockHttpServletRequest req = request(RequestMethod.POST, getURI());
+		req.setContent(json.getBytes());
+		
+		SimpleObject newPatientIdentifier = deserialize(handle(req));
+		
+		assertNotNull(PropertyUtils.getProperty(newPatientIdentifier, "uuid"));
+		assertEquals(originalCount + 1, getAllCount());
+		
+		String preferredUuid = (String) PropertyUtils.getProperty(newPatientIdentifier, "uuid");
```

Comment:
Isn't this the same call being repeated as online 135?

Reply:
On line 135, I just assert that the property is not null, then on 138, I only got the uuid (which I made sure in 135 that it is not null), used the uuid on line 139 to get the preferredIdentifier. So I don't think its a repeated call

Reply:
Are you sure PropertyUtils.getProperty(newPatientIdentifier, "uuid") is not repeated?

Reply:
Changed.